server/rest/src/main/java/org/infinispan/rest/server/operations/mediatypes/printers/HTMLOutputPrinter.java

```diff
@@ -0,0 +1,32 @@
+package org.infinispan.rest.server.operations.mediatypes.printers;
+
+import java.nio.charset.Charset;
+import java.util.Iterator;
+
+import org.infinispan.CacheSet;
+import org.infinispan.rest.server.operations.exceptions.ServerInternalException;
+import org.infinispan.rest.server.operations.mediatypes.OutputPrinter;
+
+public class HTMLOutputPrinter implements OutputPrinter {
+
+   @Override
+   public byte[] print(String cacheName, CacheSet<?> keys) {
+      StringBuilder sb = new StringBuilder("<html><body>");
+      Iterator<?> it = keys.iterator();
+      while (it.hasNext()) {
+         String key = Escaper.escapeHtml(it.next().toString());
+         sb.append(String.format("<a href=\"%s/%s\">%s</a><br/>", cacheName, key, key));
+
+      }
+      sb.append("</body></html>");
+      return sb.toString().getBytes(Charset.forName("UTF-8"));
+   }
+
+   @Override
+   public byte[] print(byte[] value) throws ServerInternalException {
+      StringBuilder sb = new StringBuilder("<html><body>");
+      sb.append(new String(value, Charset.forName("UTF-8")));
```

Comment:
Also I wonder if we want to not do UTF-8 encoding just to decode it 2 lines below. It would be a lot faster although uglier to instead just encode the surrounding tags and write the byte[] as is.